pkg/authenticator/authenticator_test.go

```diff
@@ -48,4 +52,32 @@ func TestAuthenticator(t *testing.T) {
 			So(authn.IsCertExpired(), ShouldEqual, true)
 		})
 	})
+
+	Convey("GenerateCSR", t, func() {
```

Comment:
Are those 3 different tests in one?
Why not creating 3 different tests?

Reply:
it is one test only. 
`Convey("GenerateCSR"` states the beginning of the UT for `GenerateCSR`. Then we have the preparation, and then we have 1 `Given` clause and 2 `Then` clauses which test 2 outcomes:
1. The method finishes without an error
2. The common-name is inserted in the certificate subject as expected.